### PR TITLE
ekf2: add kconfig option to disable mag, switch holybro kakutef7 to ekf2

### PR DESCRIPTION
**Pull request by dagar**
Disabling magnetometer support in ekf2 saves ~ 24.5 kB of flash.

FYI @julianoes @bkueng

### CODE REVIEW

**Comment by bkueng** [comment]
Nice, how much does it add now (bloaty ci comparison fails)?

File: src/modules/ekf2/EKF/yaw_update.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2015 Estimation and Control Library (ECL). All rights reserved.
```
Comment (bresch) [inline src/modules/ekf2/EKF/yaw_update.cpp:3]: outdated header

File: src/modules/ekf2/EKF/yaw_update.cpp [hunk 2]
```diff
@@ -0,0 +1,147 @@
+/****************************************************************************
```
Comment (bresch) [inline src/modules/ekf2/EKF/yaw_update.cpp:1]: Should we rather call this file `yaw_fusion.cpp` to follow the pattern?

**Review by bresch — COMMENTED** [review]
Nice, it also forces us to group the variables and functions together

File: src/modules/ekf2/EKF/yaw_update.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/ekf2/EKF/yaw_update.cpp]: updated

File: src/modules/ekf2/EKF/yaw_update.cpp (hunk 2 above)
Comment (dagar) [reply src/modules/ekf2/EKF/yaw_update.cpp:1]: renamed

**Review by bresch — APPROVED** [review]

**Pull request merged**